src/main/osd/osd_elements.c

```diff
@@ -2094,81 +2168,131 @@ uint8_t osdGetActiveElementCount(void)
 bool osdDrawNextActiveElement(displayPort_t *osdDisplayPort, timeUs_t currentTimeUs)
 {
     UNUSED(currentTimeUs);
+    static bool backgroundRendered = false;
     bool retval = true;
 
     if (activeElement >= activeOsdElementCount) {
         return false;
     }
 
-    if (!backgroundLayerSupported) {
+    if (!backgroundLayerSupported && !backgroundRendered) {
         // If the background layer isn't supported then we
         // have to draw the element's static layer as well.
-        osdDrawSingleElementBackground(osdDisplayPort, activeOsdElementArray[activeElement]);
-    }
+        backgroundRendered = osdDrawSingleElementBackground(osdDisplayPort, activeOsdElementArray[activeElement]);
 
-    osdDrawSingleElement(osdDisplayPort, activeOsdElementArray[activeElement]);
+        return retval;
+    }
 
-    if (++activeElement >= activeOsdElementCount) {
-        activeElement = 0;
-        retval = false;
+    // Only advance to the next element if rendering is complete
+    if (osdDrawSingleElement(osdDisplayPort, activeOsdElementArray[activeElement])) {
+        // Prepare to render the background of the next element
+        backgroundRendered = false;
+        if (++activeElement >= activeOsdElementCount) {
+            activeElement = 0;
+            retval = false;
+        }
     }
 
     return retval;
 }
 
 #ifdef USE_SPEC_PREARM_SCREEN
-void osdDrawSpec(displayPort_t *osdDisplayPort)
+bool osdDrawSpec(displayPort_t *osdDisplayPort)
 {
-    if (!ARMING_FLAG(ARMED) && osdConfig()->osd_show_spec_prearm) {
-        const uint8_t midRow = osdDisplayPort->rows / 2;
-        const uint8_t midCol = osdDisplayPort->cols / 2;
+    static enum {CLR, RPM, POLES, MIXER, THR, MOTOR, BAT, VER} specState = CLR;
+    static int currentRow;
 
-        char buff[OSD_ELEMENT_BUFFER_LENGTH] = "";
+    const uint8_t midRow = osdDisplayPort->rows / 2;
+    const uint8_t midCol = osdDisplayPort->cols / 2;
 
-        memset(buff,0,strlen(buff));
-        int len = 0;
-        int currentRow = midRow - 3;
+    char buff[OSD_ELEMENT_BUFFER_LENGTH] = "";
+
+    memset(buff, 0, strlen(buff));
```

Comment:
buff in already initialized to zero by initializer `= ""`. and strlen(buff) == 0, so no bytes are written here